Xamarin.Essentials/Calendar/Calendar.uwp.cs

```diff
@@ -0,0 +1,122 @@
+            options.FetchProperties.Add(AppointmentProperties.Duration);
+            options.FetchProperties.Add(AppointmentProperties.AllDay);
+            var sDate = startDate ?? DateTimeOffset.Now.Add(defaultStartTimeFromNow);
+            var eDate = endDate ?? sDate.Add(defaultEndTimeFromStartTime);
+
+            if (eDate < sDate)
+                eDate = sDate;
+
+            var instance = await CalendarRequest.GetInstanceAsync();
+            var events = await instance.FindAppointmentsAsync(sDate, eDate.Subtract(sDate), options);
+            var eventList = new List<DeviceEvent>();
+            foreach (var e in events)
+            {
+                if (calendarId == null || e.CalendarId == calendarId)
+                {
+                    eventList.Add(new DeviceEvent
+                    {
+                        Id = e.LocalId,
+                        CalendarId = e.CalendarId,
+                        Title = e.Subject,
+                        StartDate = e.StartTime,
+                        EndDate = !e.AllDay ? (DateTimeOffset?)e.StartTime.Add(e.Duration) : null,
+                    });
+                }
+            }
+            eventList.Sort((x, y) =>
+            {
+                if (!y.EndDate.HasValue)
+                {
+                    return 0;
+                }
+                return x.StartDate.CompareTo(y.EndDate.Value);
+            });
+
+            return eventList;
+        }
+
+        static async Task<DeviceEvent> PlatformGetEventByIdAsync(string eventId)
+        {
+            await Permissions.RequireAsync(PermissionType.CalendarRead);
+
+            var instance = await CalendarRequest.GetInstanceAsync();
+
+            Appointment e;
+            try
+            {
+                e = await instance.GetAppointmentAsync(eventId);
+                e.DetailsKind = AppointmentDetailsKind.PlainText;
+            }
+            catch (ArgumentException)
+            {
+                throw new ArgumentException($"[UWP]: No Event found for event Id {eventId}");
+            }
+            catch (NullReferenceException)
+            {
+                throw new NullReferenceException($"[UWP]: No Event found for event Id {eventId}");
+            }
+
+            return new DeviceEvent()
+            {
+                Id = e.LocalId,
+                CalendarId = e.CalendarId,
+                Title = e.Subject,
+                Description = e.Details,
+                Location = e.Location,
+                StartDate = e.StartTime,
+                EndDate = !e.AllDay ? (DateTimeOffset?)e.StartTime.Add(e.Duration) : null,
+                Attendees = GetAttendeesForEvent(e.Invitees)
+            };
+        }
+
+        static IEnumerable<DeviceEventAttendee> GetAttendeesForEvent(IEnumerable<AppointmentInvitee> inviteList)
+        {
+            var attendees = new List<DeviceEventAttendee>();
+
+            foreach (var attendee in inviteList)
```

Comment:
linq :-)

Reply:
Addressed in:
https://github.com/builttoroam/Essentials/pull/43